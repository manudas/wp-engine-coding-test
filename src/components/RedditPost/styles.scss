.reddit-post {
    width: 250px;

    display: inline-block;
    padding: 15px;
    margin: 10px;
    border: 1px solid black;
    border-radius: 5px;
    height: 250px;
    overflow: hidden;

    &_title {
        font-size: 16px;

        overflow: hidden;
        text-overflow: ellipsis;

        display: -webkit-box;
        line-clamp: 2;
        -webkit-line-clamp: 2;
        box-orient: vertical;
        -webkit-box-orient: vertical;
    }
}