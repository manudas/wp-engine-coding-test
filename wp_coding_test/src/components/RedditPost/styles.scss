@import '../globals';

.reddit-post {
    width: 250px;

    padding: 15px;
    margin: 10px;
    border: 1px solid $black;
    border-radius: 5px;
    height: 250px;
    overflow: hidden;

    display: inline-flex;
    flex-direction: column;

    &__title {
        font-size: 16px;

        overflow: hidden;
        text-overflow: ellipsis;

        display: -webkit-box;
        line-clamp: 2;
        -webkit-line-clamp: 2;
        box-orient: vertical;
        -webkit-box-orient: vertical;

        margin-top: 0;
        margin-bottom: 5px;

        flex-shrink: 0.25;
    }

    &__content {
        flex-basis: 100%;
        flex-shrink: 0.75;
        flex-grow: 0;
        overflow: hidden;

        background: linear-gradient(
            to bottom,
            transparent 0%,
            transparent 95%,
            $semiTransparentBlack 100%
        );
    }

    &__text {
        margin-top: 0;
        margin-bottom: 0;

        font-size: 14px;
    }

    &__img {
        float: right;
    }

    &__link {
        font-size: 12px;
        text-decoration: none;
        transition: all 0.3s ease-out;
        display: inline-block;
        margin-right: 0;
        margin-left: auto;

        &:hover {
            padding: 3px;
            border-radius: 3px;
            background-color: $transparentBlack;
            color: $orange;
        }
    }

    &__separator{
        border-top: 1px solid $semiTransparentBlack;
        width: 100%;
        margin: 5px 0;
    }
}
