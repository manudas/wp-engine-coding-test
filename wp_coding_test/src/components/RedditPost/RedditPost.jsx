import React from 'react';

import PropTypes from 'prop-types';

import {
    isUrl,
} from '../../utils';

import './styles.scss';

const RedditPost = ({
    title,
    thumbnail,
    text,
    url,
}) => {

    return (
        <section
            className="reddit-post"
        >
            <h2
                data-testid="postTitle"
                className="reddit-post__title"
            >
                {title}
            </h2>
            <div
                className="reddit-post__separator"
            />
            <div
                className="reddit-post__content"
            >
                {
                    // we don't render an image
                    // if no thumbnail url has been passed
                    isUrl(thumbnail)
                        ? <img
                            data-testid="postImg"
                            className="reddit-post__img"
                            src={thumbnail}
                            alt={title}
                        />
                        : null
                }
                <p
                    data-testid="postText"
                    className="reddit-post__text"
                >
                    {text}
                </p>
            </div>
            <a
                className="reddit-post__link"
                href={url}
                target="_blank"
                rel="noreferrer"
            >
                Go to post
            </a>
        </section>
    )
};

RedditPost.propTypes = {
    title: PropTypes.string,
    thumbnail: PropTypes.string,
    text: PropTypes.string,
    url: PropTypes.string.isRequired,
}

RedditPost.defaultProps = {
    title: null,
    thumbnail: null,
    text: null,
}

export default RedditPost;
