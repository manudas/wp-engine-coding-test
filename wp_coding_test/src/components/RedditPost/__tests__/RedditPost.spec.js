import React from 'react';
import {
    render,
} from '@testing-library/react';
import RedditPost from '..';

describe('RedditPost test suite', () => {

    let renderer;

    beforeEach(() => {
        renderer = render(
            <RedditPost
                title = {'Title of faked post'}
                thumbnail = {'Wrong url for a thumbnail'}
                text = {'Post text'}
                url = {'//www.reddit.com/r/reactjs/comments/lvco7l/whos_hiring_march_2021/'}
            />
        );
    });

    it('should render <RedditPost />', () => {
        const { container } = renderer;
        expect(container).toMatchSnapshot();
    });

    it('check if components inside faked post exist', async () => {
        const {
            getByTestId,
            getByText,
            queryByTestId,
        } = renderer;

        const postTitle = getByTestId('postTitle');
        const postImg = queryByTestId('postImg');
        const postText = getByTestId('postText');
        const postLink = getByText(/go to post/i);

        expect(postTitle.textContent).toBe('Title of faked post');
        expect(postImg).not.toBeInTheDocument();
        expect(postText.textContent).toBe('Post text');
        expect(postLink.href).toMatch('//www.reddit.com/r/reactjs/comments/lvco7l/whos_hiring_march_2021/');
    });
});