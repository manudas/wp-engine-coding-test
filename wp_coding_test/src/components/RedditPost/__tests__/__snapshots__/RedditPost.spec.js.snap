// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedditPost test suite should render <RedditPost /> 1`] = `
<div>
  <section
    class="reddit-post"
  >
    <h2
      class="reddit-post_title"
      data-testid="postTitle"
    >
      Title of faked post
    </h2>
    <hr
      class="reddit-post_separator"
    />
    <div
      class="reddit-post_content"
    >
      <p
        class="reddit-post_text"
        data-testid="postText"
      >
        Post text
      </p>
    </div>
    <a
      class="reddit-post_link"
      href="//www.reddit.com/r/reactjs/comments/lvco7l/whos_hiring_march_2021/"
      rel="noreferrer"
      target="_blank"
    >
      Go to post
    </a>
  </section>
</div>
`;
